src/library/Pagination/index.d.ts

```diff
@@ -1 +1,62 @@
-declare module '@targetx/mineral-ui/Pagination';
+import {
+  DetailedHTMLProps,
+  ElementType,
+  HTMLAttributes,
+  ReactElement
+} from 'react';
+
+interface PageJumperMessages {
+  label: string;
+  placeholder: string;
+}
+
+interface PageSizerMessages {
+  itemText: (pageSize: number) => string;
+  status: (
+    category: string,
+    first: number,
+    last: number,
+    total: number
+  ) => string;
+}
+
+interface PagesMessages {
+  next: string;
+  pageLabel: (
+    isCurrentPage: boolean,
+    isLastPage: boolean,
+    page: number
+  ) => string;
+  previous: string;
+}
+
+export interface PaginationMessages {
+  category?: string;
+  label: string;
+  pages?: PagesMessages;
```

Comment:
It seems like you duplicated the naming from the Flowtypes, which is fine. But they don't really make sense to me.

Also because TypeScript works slightly different to Flow, I think you'd need to export all of the interfaces recursively for this to be "correct".

An alternative might be to just embed the types in PaginationProps, e.g.
```ts
messages: {
  category?: string;
  label: string;
  pages?: {
    next: string;
    pageLabel: (
      isCurrentPage: boolean,
      isLastPage: boolean,
      page: number
    ) => string;
    previous: string;
  };
  pageJumper?: {
    label: string;
    placeholder: string;
  };
  pageSizer?: {
    itemText: (pageSize: number) => string;
    status: (
      category: string,
      first: number,
      last: number,
      total: number
    ) => string;
  };
}
```

I wouldn't normally recommend this, but, for simplicity (and avoiding the confusing naming), I think this might be the best option here.